doc/changelog.md

```diff
@@ -46,114 +139,169 @@
+    environment.
+    [(#381)](https://github.com/PennyLaneAI/catalyst/pull/381)
+
+  - The ability to dynamically release devices has been added via `DeviceReleaseOp` in the Quantum
+    MLIR dialect. This is lowered to the `__quantum__rt__device_release()` runtime instruction,
+    which updates the status of the device instance from `Active` to `Inactive`. The runtime will reuse
+    this deactivated instance instead of creating a new one automatically at runtime in a
+    multi-QNode workflow when another device with identical specifications is requested.
+    [(#381)](https://github.com/PennyLaneAI/catalyst/pull/381)
+
+  - The `DeviceOp` definition in the Quantum MLIR dialect has been updated to lower a tuple
+    of device information `('lib', 'name', 'kwargs')` to a single device initialization call
+    `__quantum__rt__device_init(int8_t *, int8_t *, int8_t *)`. This allows the runtime to initialize
+    device instances without keeping partial information of the device
+    [(#396)](https://github.com/PennyLaneAI/catalyst/pull/396)
 
 * The quantum adjoint compiler routine has been extended to support function calls that affect the
   quantum state within an adjoint region. Note that the function may only provide a single result
-  consisting of the quantum register.
-  By itself this provides no user-facing changes, but compiler pass developers may now generate
-  quantum adjoint operations around a block of code containing function calls as well as quantum
-  operations and control flow operations.
+  consisting of the quantum register. By itself this provides no user-facing changes, but compiler
+  pass developers may now generate quantum adjoint operations around a block of code containing
+  function calls as well as quantum operations and control flow operations.
   [(#353)](https://github.com/PennyLaneAI/catalyst/pull/353)
 
-* Update the `DeviceOp` definition in the Quantum MLIR dialect.
-  `DeviceOp` gets the tuple of device info: ('lib', 'name', 'kwargs')
-  and lowers the operation to one single device initialization call:
-  `__quantum__rt__device_init(int8_t *, int8_t *, int8_t *)`.
-  [(#396)](https://github.com/PennyLaneAI/catalyst/pull/396)
-
-* Add `DeviceReleaseOp` to the Quantum MLIR dialect. This will be lowered to
-  the `__quantum__rt__device_release()` runtime instruction updating the status
-  of the device instance from `Active` to `Inactive`. The runtime will reuse this
-  deactivated instance instead of creating a new one automatically at runtime in a
-  multi-qnode workflow when another device with identical specifications is requested.
-  [(#381)](https://github.com/PennyLaneAI/catalyst/pull/381)
+* `AllocOp`, `DeallocOp` have now (only) value semantics. In the frontend, the last
+  quantum register is deallocated instead of the first one. This allows to return the quantum
+  register in functions and can be given to another function (useful for quantum transformation).
```

Comment:
```suggestion
* The allocation and deallocation operations in MLIR (`AllocOp`, `DeallocOp`) now follow simple
  value semantics for qubit register values, instead of modelling memory in the MLIR trait system.
  Similarly, the frontend generates proper value semantics by deallocating the final register value.
  
  The change enables functions at the MLIR level to accept and return quantum register values,
  which would otherwise not be correctly identified as aliases of existing register values by the
  bufferization system. 
```

Reply:
thanks :)